x-pack/plugins/security/server/licensing/license_service.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { deepFreeze } from '../../../../../src/core/utils';
+import { ILicense } from '../../../licensing/server';
+import { SecurityLicenseFeatures } from './license_features';
+
+export interface SecurityLicense {
+  isEnabled(): boolean;
+  getFeatures(): SecurityLicenseFeatures;
+}
+
+export class SecurityLicenseService {
+  public setup() {
+    let rawLicense: Readonly<ILicense> | undefined;
+
+    return {
+      update(newRawLicense: Readonly<ILicense>) {
+        rawLicense = newRawLicense;
+      },
+
+      license: deepFreeze({
+        isEnabled() {
+          if (!rawLicense) {
+            return false;
+          }
+
+          const securityFeature = rawLicense.getFeature('security');
+          return (
+            securityFeature !== undefined &&
+            securityFeature.isAvailable &&
+            securityFeature.isEnabled
+          );
+        },
+
+        /**
+         * Returns up-do-date Security related features based on the last known license.
+         */
+        getFeatures(): SecurityLicenseFeatures {
```

Comment:
__note:__ we can potentially cache "features", but I don't have strong opinion on that and hence went with the simple option.